rust/sbd-server/src/lib.rs

```diff
@@ -271,102 +496,59 @@ impl SbdServer {
         task_list.push(spawn_prune_task(ip_rate.clone()));
 
         let cslot = CSlot::new(config.clone(), ip_rate.clone());
+        let weak_cslot = cslot.weak();
+
+        let app: axum::Router<()> = axum::Router::new()
+            .route("/authenticate", axum::routing::put(handle_auth))
+            .route("/{pub_key}", axum::routing::any(handle_ws))
+            .layer(axum::extract::DefaultBodyLimit::max(1024))
+            .with_state(AppState::new(
+                config.clone(),
+                ip_rate.clone(),
+                weak_cslot.clone(),
+            ));
 
-        // limit the number of connections that can be "connecting" at a time.
-        // MAYBE make this configurable.
-        // Read this as a prioritization of existing connections over incoming
-        let connect_limit = Arc::new(tokio::sync::Semaphore::new(1024));
+        let app =
+            app.into_make_service_with_connect_info::<std::net::SocketAddr>();
 
-        let mut bind_port_zero = Vec::new();
-        let mut bind_explicit_port = Vec::new();
+        let mut found_port_zero: Option<u16> = None;
 
         for bind in config.bind.iter() {
-            let a: std::net::SocketAddr = bind.parse().map_err(Error::other)?;
+            let mut a: std::net::SocketAddr =
+                bind.parse().map_err(Error::other)?;
+            if let Some(found_port_zero) = &found_port_zero {
+                if a.port() == 0 {
+                    a.set_port(*found_port_zero);
+                }
+            }
 
-            if a.port() == 0 {
-                bind_port_zero.push(a);
+            let h = axum_server::Handle::new();
+
+            if let Some(tls_config) = &tls_config {
+                let tls_config =
+                    axum_server::tls_rustls::RustlsConfig::from_config(
+                        tls_config.config(),
+                    );
+                let server = axum_server::bind_rustls(a, tls_config)
+                    .handle(h.clone())
+                    .serve(app.clone());
+                task_list.push(tokio::task::spawn(async move {
+                    let _ = server.await;
+                }));
             } else {
-                bind_explicit_port.push(a);
+                let server =
+                    axum_server::bind(a).handle(h.clone()).serve(app.clone());
+                task_list.push(tokio::task::spawn(async move {
+                    let _ = server.await;
```

Comment:
Error information we might want here?

Reply:
Guess it can't hurt: https://github.com/holochain/sbd/pull/49/commits/6b9977f2ff8e61692b3ddd3f368dc764f3326f64